#8 [Admin] CRUD Judges (members)

ochupa opened the pull request:
![](https://user-images.githubusercontent.com/74152672/163719718-e27f99d2-f10a-482c-a266-f0f766c2e05b.png)

![](https://user-images.githubusercontent.com/74152672/163719727-a5d208e8-8396-4a75-a1c1-338341b6072a.png)

![](https://user-images.githubusercontent.com/74152672/163719737-250205c1-24b4-4cca-a822-d9f6cee2c369.png)

![](https://user-images.githubusercontent.com/74152672/163719745-74bce6ee-85f9-401b-ba9d-011b48896925.png)

Commit:
[Admin] CRUD Judges (members)

Commit:
Remove duplicated enum in schema

Commit:
Add view and specs

sveredyuk commented on db/schema.rb:
```diff
@@ -10,14 +10,15 @@
   # Custom types defined in this database.
   # Note that some types may not work with other database engines. Be careful if changing database.
-  create_enum "challenge_status", ["draft", "moderation", "ready", "canceled"]
+  create_enum "challenge_status", ["draft", "moderation", "pending", "registration", "live", "complete", "canceled"]
```
please, run:
rake db:drop
rake db:create
rake db:migrate

and commit schema chamges again

ochupa replied:
Thanks. I've corrected the schema.

sveredyuk reviewed (changes requested):
Look good. Just fix schema and we can merge it

Commit:
Revert correct schema

PyrinAndrii reviewed (approved):
👍 , looks fine to me.
I think that we can merge it and extract the ['members' logic](https://github.com/DevChallengeIT/devchallenge-platform/issues/8) to a new issue. What do you all think?

Commit:
Add missed actions and removed judges/create as useless

Commit:
Add missed specs for members update/delete actions

ochupa commented:
> +1 , looks fine to me. I think that we can merge it and extract the ['members' logic](https://github.com/DevChallengeIT/devchallenge-platform/issues/8) to a new issue. What do you all think?

Thanks. It looks like I went in the wrong direction. Thank you for noticing this. I corrected my mistake and added missing actions and views mentioned in the requirement #8

sveredyuk commented on app/controllers/admin/judges_controller.rb:
```diff
@@ -0,0 +1,11 @@
+    add_breadcrumb I18n.t('resources.judges.plural'), :admin_judges_path
+
+    def index
+      @paginator, @judges = paginate Repo::Member.preload(:user, :challenge).all.judge
```
just `Repo::Member.preload(:user, :challenge).judge`

ochupa replied:
Fixed.

sveredyuk commented on app/views/admin/judges/index.html.erb:
```diff
@@ -0,0 +1,28 @@
+          <%= judge.user.email %>
+        </td>
+         <td>
+          <%= judge.challenge.title %>
```
link_to admin_challenge_members_path

ochupa replied:
Added.

sveredyuk commented on config/locales/en.yml:
```diff
@@ -61,3 +62,16 @@ en:
+      user: User
+      role: Role
+      joined: Joined
+    member:
```
just use members.singular

ochupa replied:
Fixed.

sveredyuk commented on spec/features/admin/members/remove_spec.rb:
```diff
@@ -0,0 +1,18 @@
+    click_button 'Remove'
+
+    expect(page).to have_current_path "/admin/challenges/#{challenge.slug}/members"
+    expect(page).to have_content 'Member was successfully removed'
```
expect page not to have member info

ochupa replied:
Added.

sveredyuk commented on db/migrate/20220415125558_create_members.rb:
```diff
@@ -0,0 +1,15 @@
+      t.timestamps
+    end
+
+    add_index :members, :role
```
+ index user_id, challenge_id

ochupa replied:
It looks like option `index: true` adds indices for references columns. It can be checked in [schema](https://github.com/DevChallengeIT/devchallenge-platform/pull/30/files#diff-cbf8b25d6d853acf58fa9057841f407d29ffe90649c75cf9e3f51b2d6e3aa1d3R83)

PyrinAndrii commented on app/controllers/admin/members_controller.rb:
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Admin
+  class MembersController < BaseController
```
I see that we don't have it in the issue, but I think will be great to add `new` & `create` action. Let's add it?

sveredyuk replied:
No need need this right now

PyrinAndrii commented:
@ochupa, also I think makes sense to add a few members to seeds. Could we do that?
